x-pack/plugins/canvas/canvas_plugin_src/renderers/shape/index.js

```diff
@@ -39,22 +39,30 @@ export const shape = () => ({
       const height = domNode.offsetHeight;
 
       // adjust viewBox based on border width
-      let [minX, minY, maxX, maxY] = initialViewBox;
+      let [minX, minY, shapeWidth, shapeHeight] = initialViewBox;
+      const borderOffset = Math.max(borderWidth, 0) / 2;
 
-      const xScale = (maxX - minX) / width;
-      const yScale = (maxY - minY) / height;
-      const borderOffset = borderWidth / 2;
-      const xOffset = borderOffset * xScale;
-      const yOffset = borderOffset * yScale;
+      if (width) {
```

Comment:
Works fine! We could avoid the introduction of the conditional and the `else` branch with
```
        const xScale = width ? (shapeWidth - minX) / width : 0;
        const yScale = height ? (shapeHeight - minY) / height : 0;
```

Reply:
I'm not sure that'll work in this case. I need to both decrease `minX` and increase `shapeWidth` if the element has a width > 0. Otherwise, I want `shapeWidth` to be 0 and same for height. I need to the `xOffset` value for adjusting the `minX` and `shapeWidth`.